ArtOfIllusion/src/artofillusion/script/GroovyScriptEngine.java

```diff
@@ -74,9 +74,10 @@ public void executeScript(String scriptBody, Map<String, Object> variables) thro
         script.run();
         
     }
-    catch (CompilationFailedException e)
-    {
-      throw new ScriptException(e.getMessage(), -1);
+    catch(GroovyRuntimeException gre) {
+      Throwable tt = StackTraceUtils.sanitize(gre);
+      int ln = tt.getStackTrace()[0].getLineNumber() - numImports;
```

Comment:
What's up with this line? What happens if there are imports in the actual script body?

Reply:
This works ok. Idea is to shift line counter on numImports lines 
Imports from script code is not counted to numImports